pallets/schemas/src/lib.rs

```diff
@@ -625,19 +1016,57 @@ pub mod pallet {
 		/// Retrieve a schema info by id
 		pub fn get_schema_info_by_id(schema_id: SchemaId) -> Option<SchemaInfoResponse> {
 			if let Some(schema_info) = SchemaInfos::<T>::get(schema_id) {
-				let saved_settings = schema_info.settings;
-				let settings = saved_settings.0.iter().collect::<Vec<SchemaSetting>>();
 				let response = SchemaInfoResponse {
 					schema_id,
+					intent_id: schema_info.intent_id,
```

Comment:
Make sure this goes at the end so SCALE decodes properly even for the old struct